### ISSUE TITLE
IO::TimeoutError exceptions are not handled, breaking automatic retry

### ISSUE THREAD

**Issue by skaes** [petergoldstein/dalli#967]
In Ruby 3.2.0, sockets can raise IO::TimeoutError when blocking operations time out ([Ruby 3.2.0 News](https://github.com/ruby/ruby/blob/ruby_3_2/NEWS.md), [feature 18630](https://bugs.ruby-lang.org/issues/18630)) . These happen on read/write operations in Dalli. Dalli only rescues Timeout::Error.

The overall effect is that operations that time out are never retried. Instead they just log an error:

```
Unexpected exception during Dalli request: IO::TimeoutError: Blocking operation timed out!
```
I'm not sure why this happens, because from the Ruby NEWS file one would assume this only happens when the timeout attribute is set on the IO instance and I could not find it in the Dalli code.

**Comment by skaes** [comment]
https://github.com/petergoldstein/dalli/pull/968 is an attempt to improve the situation.

**Comment by petergoldstein** [comment]
This is merged and published in 3.2.6

**Comment by skaes** [comment]
@petergoldstein would you mind tagging the release?
